### ISSUE TITLE
No borders, no waitron commands work on Cinnamon

### ISSUE THREAD

**Issue by ohmree** [tudurom/windowchef#35]
I took [this](https://github.com/Gigadoc2/i3-cinnamon), substituted all occurrences of `i3` with `windowchef` and fixed the PKGBUILD md5sums.  
It works, I can open some programs through `sxhkd` keybinds and the focus follows my mouse but non of the `waitron` commands work. Not in the terminal and not through `sxhkd`. I also don't get any window borders.

I have no idea why this is happening, but I guess that Cinnamon is interfering with windowchef somehow.
I'm on Antergos, using Cinnamon version `3.4.4` and windowchef version `0.3.2` if that helps.

Here's a scrot:  
![image](https://user-images.githubusercontent.com/13455401/28588517-822b202c-7183-11e7-983a-6a5f427892b0.png)

Note that I can grab the part that the black arrow is pointing at to move the window but I think it's Firefox behavior and not directly related to windowchef:
![image](https://user-images.githubusercontent.com/13455401/28588695-0e5dab6e-7184-11e7-988d-9a128861665a.png)

**Comment by tudurom** [comment]
does waitron return any error? i'm not at my computer right now, but i suspect that windowchef isn't running at all in your case. have you checked that?

**Comment by ohmree** [comment]
No errors. `waitron wm_config <some_option>` segfaulted but I don't know if
it's supposed to print the value of `<some_option>` if no more arguments
are given. If it isn't and there's no proper check (if `argc` is not the
amount I want it to be  a then exit with error status and print usage) then
it should segfault (AFAIK).

And windowchef is definitely running, `ps aux | grep windowchef` shows that
it is indeed running (there's `windowchef` and `[windowchef] <defunct>`, I
have no idea what the second one is, maybe it somehow gets started twice
because of an error in the session file/scripts?)

**Comment by tudurom** [comment]
instead of testing waitron with the `wm_config` command (yes it is expected to segfault when no arguments are given), try moving the window.

**Comment by tudurom** [comment]
also the second defunct process is normal, it doesn't do anything bad, i will make it disappear someday

**Comment by ohmree** [comment]
I did try moving the window, it doesn't move anything and doesn't print
anything.

**Comment by ohmree** [comment]
![image](https://user-images.githubusercontent.com/13455401/28595123-2191a96c-719b-11e7-967f-a8bcd11aa8b6.png)  
Ok this is weird. I tried running some `cinnamon-*` apps so I could get a bar (When I had KDE + bspwm I think I just installed [this](https://aur.archlinux.org/packages/bspwm-kde-session/) and the bar worked automagically).  `cinnamon-settings` opened the settings menu as expected, but `cinnamon-launcher` (I thought it's an application launcher, turns out it starts a Cinnamon session) just started the bar and gave me Cinnamon's borders on windowchef's windows. As you can see, muffin is not running and windowchef is.  
`alt-tab` (which I believe is managed by muffin) doesn't work (although `meta + arrow-key` does tile windows like it does in regular Cinnamon, perhaps because windows aren't tied to a specific WM so as long as something is written for X11 and uses the proper procedures it can move windows?).  
There's no context menu or icons on the desktop but Cinnamon's widgets are there, and `meta` opens up the application launcher ("start menu"). Maybe I have to completely disable Cinnamon's shortcuts? (I know you have to do this if you want to have XFCE with bspwm)

So to sum this up I ended up with a hybrid desktop that has none of the functions that I want from windowchef, I wonder why that is. I'll try doing the same in a VM and maybe installing bspwm on this machine and see if the problem persists, this is really peculiar!

EDIT: It also seems that windows now only focus on click, it wasn't like this before I executed `cinnamon-launcher`. Very weird

EDIT 2: After playing with it some more I noticed that I can change the window effects through Cinnamon's settings and it, well, actually worked. So maybe muffin is running somehow even though nothing is showing that it's running? Also sxhkd shortcuts don't work even though sxhkd is running, so I think the cinnamon session is interfering with that.

**Comment by ohmree** [comment]
Ok, so bspwm actually works (although it makes `cinnamon-launcher` crash when I try to execute it).  
In order to get the bspwm session to show up on my DM I did the same thing as before but with bspwm instead of windowchef (cloned [this](https://github.com/Gigadoc2/i3-cinnamon), replaced all occurrences of `i3` with `bspwm`, fixed md5sums in the PKGBUILD, ran `makepkg && sudo pacman -U name-of-package-version-arch.pkg.tar`).

scrot:  
![image](https://user-images.githubusercontent.com/13455401/28596780-3180864c-71a3-11e7-9694-ee916b4fbdf1.png)

**Comment by tudurom** [comment]
Closing this because it's an issue with Cinnamon. Most probably it's doing stuff to the root window and doesn't let the client messages from `waitron` to get received by the wm.
